### PR TITLE
implement copying of resources as in the UI

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: tower_cli/models/base.py [hunk 1]
```diff
@@ -1215,6 +1215,22 @@ def create(self, **kwargs):
...
+        orig = self.read(pk, fail_on_no_results=True,
+                         fail_on_multiple_results=True)
+        orig = orig['results'][0]
+        # Remove default values (anything where the value is None).
+        self._pop_none(kwargs)
+
+        newresource = copy(orig)
+        newresource.update(kwargs)
+        newresource['id'] = None
+        newresource['name'] += " @ %s" % (time.strftime('%X'))
```
Comment (AlanCoding) [inline tower_cli/models/base.py]: Slight nuanced case - making a copy of a copy. Try using this instead:

```python
newresource['name'] = newresource['name'].split('@', 1)[0] + ' @ ' + time.strftime('%I:%M:%S %p')
```

Let me know if that doesn't work for you. Stylistically you might want to alter it to use the `%` synatx.

File: tower_cli/models/base.py [hunk 2]
```diff
@@ -1215,6 +1215,22 @@ def create(self, **kwargs):
...
+        """
+        orig = self.read(pk, fail_on_no_results=True,
+                         fail_on_multiple_results=True)
+        orig = orig['results'][0]
+        # Remove default values (anything where the value is None).
+        self._pop_none(kwargs)
+
+        newresource = copy(orig)
+        newresource.update(kwargs)
+        newresource['id'] = None
```
Comment (AlanCoding) [inline tower_cli/models/base.py]: Maybe just `.pop('id')` instead?

File: tower_cli/models/base.py [hunk 3]
```diff
@@ -1215,6 +1215,22 @@ def create(self, **kwargs):
...
+        """Copy an object.
+        """
+        orig = self.read(pk, fail_on_no_results=True,
+                         fail_on_multiple_results=True)
+        orig = orig['results'][0]
+        # Remove default values (anything where the value is None).
+        self._pop_none(kwargs)
+
+        newresource = copy(orig)
+        newresource.update(kwargs)
```
Comment (AlanCoding) [inline tower_cli/models/base.py]: This over-ride functionality is meaningful for properties not in the identity tuple, I think that's fine but not sure how to document. I'd mention in the command's help text that it applies user-provided params to the _new_ object.

File: tower_cli/models/base.py (hunk 2 above)
Comment (evgeni) [inline tower_cli/models/base.py]: ack

File: tower_cli/models/base.py (hunk 3 above)
Comment (evgeni) [inline tower_cli/models/base.py]: done

File: tower_cli/models/base.py (hunk 1 above)
Comment (evgeni) [inline tower_cli/models/base.py]: yeah, good catch.

any reason you used explicit `time.strftime('%I:%M:%S %p')` and not `%X` which should (at least sometimes) match the user locale?

Comment (AlanCoding) [inline tower_cli/models/base.py]: The main reason for that was to have the am/pm for purely aesthetic reasons.

Comment (AlanCoding) [inline tower_cli/models/base.py]: If `%X` is more internationally compliant, I would say leave it as is.

**Pull request merged**